homeassistant/components/homekit/util.py

```diff
@@ -385,6 +385,32 @@ def density_to_air_quality(density):
     return 5
 
 
+def density_to_air_quality_pm10(density):
+    """Map PM10 density to HomeKit AirQuality level."""
+    if density <= 40:
+        return 1
+    if density <= 80:
+        return 2
+    if density <= 120:
+        return 3
+    if density <= 300:
+        return 4
+    return 5
+
+
+def density_to_air_quality_pm25(density):
```

Comment:
There is a `density_to_air_quality_pm25` above that has the docstring `"""Map PM2.5 density to HomeKit AirQuality level."""` above but provides a different mapping.

Reply:
The `density_to_air_quality` is the original mapping function for the original`AirQualitySensor` class. The function should be mapped with a PM 2.5 value. I found a new mapping scale for [PM2.5](https://www.epa.vic.gov.au/for-community/environmental-information/air-quality/pm25-particles-in-the-air)and [PM10](https://www.epa.vic.gov.au/for-community/environmental-information/air-quality/pm10-particles-in-the-air). That's why I create a new mapping method for the PM2.5 and PM10.

If we remove the  `density_to_air_quality`, we could not retain the `AirQualitySensor` class for a generic air sensor. According to the Apple homekit menu, a homekit AirQualitySensor accessory class must provides a AIR_QUALITY value.